Fails to build with Clang

Regressed by #4465, see [error log](https://github.com/yuzu-emu/yuzu/files/5016588/yuzu-qt5-s20200803.log).
```
$ c++ --version
FreeBSD clang version 10.0.1 (git@github.com:llvm/llvm-project.git llvmorg-10.0.1-0-gef32c611aa2)
Target: x86_64-unknown-freebsd13.0
Thread model: posix
InstalledDir: /usr/bin

$ cmake -B /tmp/yuzu_build -G Ninja
$ cmake --build /tmp/yuzu_build
FAILED: src/common/CMakeFiles/common.dir/atomic_ops.cpp.o
/usr/bin/c++  -DARCHITECTURE_x86_64=1 -DBOOST_ALL_NO_LIB -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_NO_LIB -DFMT_SHARED -DNDEBUG -DXBYAK_NO_OP_NAMES -I/path/to/yuzu/src/. -I/path/to/yuzu/externals/./microprofile -isystem /usr/local/include/SDL2 -isystem /usr/local/include -isystem /path/to/yuzu/externals/./xbyak/xbyak -O3 -DNDEBUG   -Wall -Werror=implicit-fallthrough -Werror=missing-declarations -Werror=reorder -Wextra -Wmissing-declarations -Wno-attributes -Wno-unused-parameter -fconcepts -mcx16 -std=gnu++2a -MD -MT src/common/CMakeFiles/common.dir/atomic_ops.cpp.o -MF src/common/CMakeFiles/common.dir/atomic_ops.cpp.o.d -o src/common/CMakeFiles/common.dir/atomic_ops.cpp.o -c /path/to/yuzu/src/common/atomic_ops.cpp
c++: error: unknown argument: '-fconcepts'
```

Note, on FreeBSD using GCC isn't easy. Clang defaults to libc++ but GCC defaults to libstdc++. Dependencies (e.g., Qt5) are built by default compiler thus linked against libc++. Mixing C++ libraries (libc++ with libstdc++) leads to segfaults unless built against the same runtime (libsupc++, libc++abi, libcxxrt). GCC can be coerced to use libc++ but it's fragile e.g.,

```
$ export CC=gcc10 CXX=g++10 CXXFLAGS='-nostdinc++ -isystem/usr/include/c++/v1' LDFLAGS=-Wl,-rpath,/usr/local/lib/gcc10
[...]
FAILED: bin/yuzu
[...]
/usr/local/bin/ld: src/video_core/libvideo_core.a(gl_texture_cache.cpp.o): undefined reference to symbol '_ZNSt3__19to_stringEj'
/usr/local/bin/ld: /usr/lib/libc++.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -60,8 +60,13 @@ else()
         -Wmissing-declarations
         -Wno-attributes
         -Wno-unused-parameter
-        -fconcepts
     )
+
+    # TODO: Remove when we update to a GCC compiler that enables this
+    #       by default (i.e. GCC 10 or newer).
+    if (CMAKE_CXX_COMPILER_ID STREQUAL GNU)
+        add_compile_options(-fconcepts)
+    endif()
 
     if (ARCHITECTURE_x86_64)
         add_compile_options("-mcx16")

